src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelGene.tsx

```diff
@@ -68,18 +67,55 @@ const TrackPanelGene = (props: TrackPanelGeneProps) => {
       trackId: GENE_TRACK_ID
     })
   );
-  const { toggleTrack, genomeBrowser } = useGenomeBrowser();
-
+  const browserGenomeTrackStates = useAppSelector(
+    getBrowserActiveGenomeTrackStates
+  );
+  const { toggleTrack, updateTranscriptTracks, genomeBrowser } =
+    useGenomeBrowser();
   const dispatch = useAppDispatch();
 
+  const transcriptTrackStatuses =
+    browserGenomeTrackStates?.transcriptTracks?.[focusObjectId] ?? {};
+  let sortedTranscripts: TrackPanelTranscriptType[] = [];
+
   useEffect(() => {
     toggleTrack({ trackId: GENE_TRACK_ID, status: trackStatus });
   }, [genomeBrowser]);
 
+  useEffect(() => {
+    if (sortedTranscripts.length) {
+      const visibleTranscriptIds = sortedTranscripts
+        .map((transcript) => transcript.stable_id)
+        .filter((transcriptId) => {
+          if (
+            transcriptTrackStatuses[transcriptId] === Status.SELECTED ||
+            transcriptTrackStatuses[transcriptId] === undefined
+          ) {
+            return true;
+          } else {
+            return false;
+          }
```

Comment:
Nitpick; but this function can just return what is being checked in the if-condition